File: hrdj/hrdj.py
import os
import execjs
import requests
import json

os.environ["NODE_PATH"] = os.path.join('../' + "/node_modules")


class HRDJ:
    def __init__(self):
        self.__js = self.__js_load()
        self.token = None

        self.header = {
            "Accept": "application/json, text/plain, */*",
            "Accept-Encoding": "gzip, deflate, br",
            "Accept-Language": "zh-CN,zh;q=0.9,en;q=0.8,en-GB;q=0.7,en-US;q=0.6,ja;q=0.5",
            "Cache-Control": "no-cache",
            "Connection": "keep-alive",
            "Content-Length": "137",
            "Content-Type": "application/json",
            "Host": "user.hrdjyun.com",
            "Origin": "https://www.hh1024.com",
            "Pragma": "no-cache",
            "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/107.0.0.0 Safari/537.36 Edg/107.0.1418.56"
        }

    def __js_load(self):
        with open('./hrdj.js') as f:
            js_code = f.read()

        return execjs.compile(js_code)

    def __get_form(self, account: str, pswd: str) -> dict:
        return self.__js.call('getForm', account, pswd)

    def __post(self, api, header=None, data=None, params=None):
        if not header:
            header = self.header

        res = requests.post(api, headers=header, data=data, params=params)

        if res.status_code == 200:
            res_json = res.json()
            if res_json['status'] == 0:
                return {'status': 0, 'msg': 'OK', 'data': res_json['data']}
            else:
                return {'status': 1, 'msg': f'status:{res_json["status"]}-{res_json["message"]}', 'data': {}}
        else:
            return {'status': 2, 'msg': f'{res.status_code}-请求失败', 'data': {}}

    def login(self, account: str, pswd: str) -> str:
        api = 'https://user.hrdjyun.com/wechat/phonePwdLogin'

        # res = requests.post(url=api, data=json.dumps(self.__get_form(account, pswd)), headers=self.header)
        res = self.__post(api, data=json.dumps(self.__get_form(account, pswd)))
        if res['status'] == 0:
            self.token = res['data']['token']
            return '登录成功'
        else:
            return '登录失败'

    def get_items(self, date) -> dict | list:
        if self.token:
            api = 'https://ucp.hrdjyun.com:60359/api/dy'

            form = {
                'param': self.__js.call('getParams',date),
                'sign': self.__js.call('getSign', date),
                'tenant': '1',
                'token': self.token,
                'timestamp': str(self.__js.call('getTime'))
            }
            res = self.__post(api, data=json.dumps(form))

            if res['status'] == 0:
                return res['data']
            else:
                return res
        else:
            return {'status':3,'msg':'未登录'}

if __name__ == '__main__':
    hr = HRDJ()
    #hr.login('15375250723', 'p15375250723')
    hr.token = 'OpqHUDy9nTkJG6HK3em4bZwh6bcj+4Xcr8kIvDymrZdzdibcvngdAQ=='
    i = hr.get_items('2022-11-30')
    print(i)

File: hrdj/hrdj.js
const crypto = require('crypto-js')

function getTime(){
    return (new Date()).getTime()
}

function P(n) {
    var e = [], t = "";
    for (var a in n)
        e.push(n[a]);
    for (var i = 0; i < e.length; i++)
        t += e[i] + "";
    return t += "JzyqgcoojMiQNuQoTlbR5EBT8TsqzJ", t
}

function S(n) {
    for (var e = Object.keys(n).sort(), t = {}, a = 0; a < e.length; a++)
        t[e[a]] = n[e[a]];
    return t
}

function md5(text){
    return crypto.MD5(text).toString()
}

function getSig(e){
    return md5(P(S(e)))
}

function k(text){
    return crypto.SHA256(text).toString()
}

function E(n, e) {
    return k("param=" + JSON.stringify(n) + "&timestamp=" + e + "&tenant=1&salt=" + 'kbn%&)@<?FGkfs8sdf4Vg1*+;`kf5ndl$')
}

function getParams(date) {
    params = {
        no:'dy009',
        data:{
            categoryName: "",
            days:1,
            dyCatName:"",
            startDay:date,
            type:1
        }
    }
    return params
}

function getSign(date){
    params = getParams(date)
    return E(params,new Date().getTime())
}

function getForm(account,pswd){
    let form = {
        phoneNum: account,
        pwd: md5(pswd),
        t: new Date().getTime().toString(),
        tenant: 1
    }

    form.sig = getSig(form)

    return form
}

console.log(E('{"no":"dy0009","data":{"days":1,"type":1,"dyCatName":"","categoryName":"","startDay":"2022-11-30"}}',1669875616353))
